### PR TITLE
Feature/main menu

### PR DESCRIPTION
**Pull request by MaDud**
Create basic structure for menu with toggle btn for mobile view.

### CODE REVIEW

**Review by kingam280 — APPROVED** [review]

**Review by kingam280 — APPROVED** [review]

File: src/index.js [hunk 1]
```diff
@@ -1,8 +1,10 @@
-import 'regenerator-runtime/runtime' //async/await with Parcel
+import 'regenerator-runtime/runtime'; //async/await with Parcel
 import {App} from "./app/App";
 
 const ONE_SECOND_MILLIS = 1000;
 const SW_API_BASE_URL = process.env.SW_API_BASE_URL || "https://swapi.dev/api";
 const QUIZ_MAX_TIME = process.env.QUIZ_MAX_TIME_SECONDS ? process.env.QUIZ_MAX_TIME_SECONDS * ONE_SECOND_MILLIS : 120 * ONE_SECOND_MILLIS;
```
Comment (radekwojpl2) [inline src/index.js:6]: do we need this variable ?

File: src/index.js [hunk 2]
```diff
@@ -1,8 +1,10 @@
-import 'regenerator-runtime/runtime' //async/await with Parcel
+import 'regenerator-runtime/runtime'; //async/await with Parcel
 import {App} from "./app/App";
 
 const ONE_SECOND_MILLIS = 1000;
 const SW_API_BASE_URL = process.env.SW_API_BASE_URL || "https://swapi.dev/api";
```
Comment (radekwojpl2) [inline src/index.js:5]: same here, do we need that ?

File: src/app/MainMenu.js [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+  backdropForSearch.appendChild(modalForSearch);
+
+  //APPEND MENU AND SEARCH STRUCTURE TO PAGE
+  const placeToAppend = document.getElementById('swquiz-app');
+  document.body.insertBefore(menu, placeToAppend);
+  document.body.insertBefore(backdropForSearch, placeToAppend)
+
+  //SEARCH
+  //data to connect with spoonacular
+  const API_KEY = 'a69c65ede3bb4ac3b262c5b425b4f835';
```
Comment (radekwojpl2) [inline src/app/MainMenu.js:82]: Every single feature should have its own API key, look at it on next PR

File: src/app/MainMenu.js [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+    backdrop.style.opacity = 1;
+    backdrop.style.zIndex = 100;
+
+    //create information about search value in modal
+    const searchInfo = `Results for search: ${inputForSearch.value}`
+    const searchTitle = document.createElement('h2');
+    searchTitle.innerText = searchInfo;
+    resultsSection.appendChild(searchTitle);
+
+    sendRequest(inputForSearch)
```
Comment (radekwojpl2) [inline src/app/MainMenu.js:162]: I think you miss try/catch block here, send request will throw error on failure, then this error should be logged, we can think about it next time

File: src/app/MainMenu.js [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+  //TOOGLE MENU
+  const toggleBtn = document.querySelector('.navigationBtn');
+  const navigationBox = document.querySelector('.navigationBox')
+
+  //change visibility of menu for mobile
+  const changeVisibilityForMenu = () => {
+    navigationBox.classList.toggle('navigationBoxHidden');
+  }
+
+  toggleBtn.addEventListener('click', e => {
```
Comment (radekwojpl2) [inline src/app/MainMenu.js:191]: I would avoid one-letter variables,  even when the variable is obvious

**Comment by mateugazi** [comment]
I don't see any test files (I'm not sure if they should be committed, but I think it would be nice to see how you test it, because somebody could create the test that could find some error)

**Review by mateugazi — CHANGES_REQUESTED** [review]
I couldn't find invoking of the function 'MainMenu()' from 'MainMenu.js' in 'index.html', so I wondered if it's true and cloned this branch and test it locally and the 'index.html' was left blank. I had to create additional file which invoke this function here and after that the navbar appeared properly. Did I miss something or am I right?

**Comment by MaDud** [comment]
> I couldn't find invoking of the function 'MainMenu()' from 'MainMenu.js' in 'index.html', so I wondered if it's true and cloned this branch and test it locally and the 'index.html' was left blank. I had to create additional file which invoke this function here and after that the navbar appeared properly. Did I miss something or am I right?

I remove it from index.html as I was using this file for test for 'MainMenu.js. I am not sure if it should be implemented on page which we don't use - if this will be place for our main page then of course it should be implemented by owner of this card. Now it is possible to see invoking of MainMenu function in recipe.js and at page './recipe.html'

**Review by mateugazi — APPROVED** [review]

**Comment by mateugazi** [comment]
Oh now I see, then sorry for my inappropriate question. Everything looks good to me so you have my approval for this pr. Good job :)

**Review by kingam280 — APPROVED** [review]

**Review by JotGabiga — APPROVED** [review]
Great job!

**Pull request merged**